src/pages/preview/[lang]/services/[slug].astro

```diff
@@ -0,0 +1,80 @@
+---
+import Base from "@layouts/Base.astro";
+import { serviceQuery } from "@api/queries";
+import { imageBuilder, loadQuery } from "@api/sanity";
+import SanityComponentsRenderer from "@components/cms-components/SanityComponentsRenderer.astro";
+import { type ServiceWithReferences } from "src/sanity/custom.sanity.types";
+import HeroHeader from "@components/uicomposition/HeroHeader.astro";
+import Customers from "@components/uicomposition/Customers.astro";
+import Faq from "@components/uicomposition/Faq.astro";
+import FaqQuestion from "@components/uielements/FaqQuestion.astro";
+import { PortableText } from "astro-portabletext";
+
+export const prerender = false;
+const { lang, slug } = Astro.params;
+if (!lang || !slug) {
+  return Astro.redirect("/404");
+}
+
+const { data: servicePage } = await loadQuery<ServiceWithReferences>({
+  query: serviceQuery,
+  params: { language: lang, slug: slug },
+});
+---
+
+<Base
+  title={servicePage?.metadata?.seo?.title || servicePage?.intro?.title!}
+  description={servicePage?.metadata?.seo?.description || ""}
+  language={lang.toLowerCase()}
+  slug={slug}
+  preview={true}
+>
+  <HeroHeader
+    headline={servicePage.intro?.title}
```

Comment:
if servicepage is null this throws an error: 
![image](https://github.com/user-attachments/assets/755461ac-0933-4e5e-b928-2bf4c5009320)
Not sure if it's something we catch or just send all errors to 404's or something